src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -1284,6 +1344,19 @@
                    :state (cond-> state column-validation-message (conj column-validation-message))
                    :cut true})
 
+                [["subquery" relationship expr]]
+                (let [subquery-expr (push-down-context
+                                     (user-query->logical-obj (str "select_" relationship))
+                                     expr)
+                      nested-qc (:query-context (meta subquery-expr))]
+
+                  {:node (vary-meta ["subquery" relationship
+                                     (vary-meta expr
+                                                assoc :query-context nested-qc)]
+                                    assoc :query-context context)
+                   :state state
+                   :cut true})
```

Comment:
It's a hint to the zipper functions. This code is a pre-order traversal, meaning the parent in the tree is visited first, then the children. The cut tells the zipper code it should not visit the children of this node. What this code does is attach a query context to each of the expressions in the query. A query is issued for the reports endpoint, but nothing in the JSON query expression says anything about reports. So this code walks the tree and attaches a query context to each piece of that query so it can be processed in isolation. A subquery shouldn't be annotated with reports, but rather events (or whatever was specified in the query). So it doesn't want to walk into that branch of the query and instead, do something different at that part.